Userland/Services/SystemServer/main.cpp

```diff
@@ -492,15 +510,21 @@ ErrorOr<int> serenity_main(Main::Arguments arguments)
     event_loop.register_signal(SIGCHLD, sigchld_handler);
     event_loop.register_signal(SIGTERM, sigterm_handler);
 
+    // We need this variable to see if we need to start up the shell or not
+    auto single_app = determine_single_app_mode();
+
     // Read our config and instantiate services.
     // This takes care of setting up sockets.
     auto config = (user)
         ? TRY(Core::ConfigFile::open_for_app("SystemServer"))
         : TRY(Core::ConfigFile::open_for_system("SystemServer"));
     for (auto const& name : config->groups()) {
-        auto service = TRY(Service::try_create(*config, name));
-        if (service->is_enabled())
-            g_services.append(move(service));
+        if ((name == "Taskbar" and single_app != "no") or (name == "Desktop" and single_app != "no") or (name == "Terminal" and single_app != "no") or (name == "SingleAppDaemon" and single_app == "no")) {
```

Comment:
Not very problematic, but `and` and `or` aren't used anywhere else in the system